swift/apple/FirezoneNetworkExtension/NetworkSettings.swift

```diff
@@ -166,6 +154,23 @@ class NetworkSettings {
       completionHandler?(error)
     }
   }
+
+  // See https://github.com/firezone/firezone/issues/2939
+  // 1. Read existing system default servers from resolv.conf
+  // 2. Remove our sentinel if it's already there
+  // 3. Create a new Array with our sentinel first (order matters)
+  private func createDNSSettings(dnsSentinelAddress: String) -> NEDNSSettings {
+    var systemDefaultResolvers = Resolv().getservers().map(Resolv.getnameinfo)
+    systemDefaultResolvers.removeAll(where: { dnsSentinelAddress == $0 })
+    let dnsServers = [dnsSentinelAddress] + systemDefaultResolvers
+    let dnsSettings = NEDNSSettings(servers: dnsServers)
```

Comment:
I don't think we should be adding pre-existing DNS servers as the tunnel's DNS servers. That would not be a solution we would want to trust / maintain going forward.

I'd go with one of these alternatives:
  - Use what scutils does internally, if we can, or
  - Explore using the path monitor to keep track of the primary DNS server and return that